### PR TITLE
Adding Ruby Getting Started Guide

### PR DESCRIPTION
**Pull request by erikaheidi**
This PR includes the Ruby Getting Started Guide that I've been working on. I got some help from @found-it and was able to get unblocked and finish the guide :)

The demos are available in the edu-images-demo repository: https://github.com/chainguard-dev/edu-images-demos/tree/main/ruby

**Preview:** https://deploy-preview-679--ornate-narwhal-088216.netlify.app/chainguard/chainguard-images/reference/ruby/getting-started-ruby/

### CODE REVIEW

File: content/chainguard/chainguard-images/reference/ruby/getting-started-ruby.md [hunk 1]
```diff
@@ -0,0 +1,322 @@
...
+end
+
+```
+
+Copy this code to your `inky.rb` script, then save and close the file.
+
+Next, pull down the ASCII `inky.txt` file with `curl`. You can [inspect the URL](https://raw.githubusercontent.com/chainguard-dev/edu-images-demos/main/ruby/inky-says/inky.txt) before downloading it to ensure it is safe to do so. Make sure you are still in the same directory where your `inky.rb` script is.
+
+```shell
+curl -O https://raw.githubusercontent.com/chainguard-dev/edu-images-demos/main/ruby/octo-facts/facts.txt
```
Comment (jamonation) [inline content/chainguard/chainguard-images/reference/ruby/getting-started-ruby.md:209]:
```suggestion
curl -O https://raw.githubusercontent.com/chainguard-dev/edu-images-demos/main/ruby/inky-says/inky.txt
```

**Review by jamonation — APPROVED** [review]
Couple suggestions here and there, nothing major. I love the examples, they're so whimsical, they make building container images fun!

I made the image names explicit throughout so that users don't mistakenly pull a docker hb image and get incorrect output. Not sure if you want to link to the overview page for them anywhere else, or just in the intro like I suggested.

File: content/chainguard/chainguard-images/reference/ruby/getting-started-ruby.md [hunk 2]
```diff
@@ -0,0 +1,322 @@
...
+
+With the demo ready, you can now set up a Dockerfile to build a custom image for your Ruby application. Make sure you're still on the same directory as the application files, then create a new Dockerfile using your text editor of choice:
+
+```shell
+nano Dockerfile
+```
+
+The following Dockerfile will:
+
+1. Start a new image based on the `ruby:latest` image;
```
[the remark this answers is not in the document]
Comment (erikaheidi) [reply content/chainguard/chainguard-images/reference/ruby/getting-started-ruby.md:106]: good catch!

File: content/chainguard/chainguard-images/reference/ruby/getting-started-ruby.md [hunk 3]
```diff
@@ -0,0 +1,322 @@
...
+toc: true
+---
+
+The Ruby images maintained by Chainguard are a mix of development and production distroless images that are suitable for building and running Ruby workloads.
+
+Because Ruby applications typically require the installation of third-party dependencies via [Rubygems](https://rubygems.org/), using a pure distroless image for building your application would not work. In cases like this, you'll need to implement a [multi-stage Docker build](https://docs.docker.com/build/building/multi-stage/) that uses one of the `-dev` images to set up the application.
+
+In this guide, we’ll cover two examples to showcase Ruby container images based on Wolfi as a runtime. In the first, we’ll use the minimal image containing just Ruby to execute a demo that doesn't have any external dependencies. In the second example, we'll set up a multi-stage Docker build to run a demo that requires the installation of Rubygems via `bundler`.
+
+{{< details "What is distroless" >}}
```
[the remark this answers is not in the document]
Comment (erikaheidi) [reply content/chainguard/chainguard-images/reference/ruby/getting-started-ruby.md:23]: That makes sense, I have tried that here but I don't really like how it kinda breaks the flow of the introduction. The three blurbs together at the bottom look more like a little glossary so it doesn't cause the same break, I think I will keep this way for consistency with the other guides.

**Pull request merged**